src/aws/auth.rs

```diff
@@ -266,8 +267,20 @@ impl AwsAuthentication {
                 ..
             } => {
                 let auth_region = region.clone().map(Region::new).unwrap_or(service_region);
-                let mut builder =
-                    AssumeRoleProviderBuilder::new(assume_role).region(auth_region.clone());
+
+                let tls_settings = MaybeTlsSettings::tls_client(tls_options)?;
+                let tls_connector = build_tls_connector(tls_settings)?;
+
+                let connector = HyperClientBuilder::new().build(tls_connector);
+                let config = SdkConfig::builder()
+                    .http_client(connector)
+                    .region(auth_region.clone())
+                    .time_source(SystemTimeSource::new())
+                    .build();
+
+                let mut builder = AssumeRoleProviderBuilder::new(assume_role)
+                    .region(auth_region.clone())
+                    .configure(&config);
```

Comment:
Do we need to add this to the `AwsAuthentication::Default ` arm too? The default credential provider can also assume role using STS based on environment variable configuration (`$AWS_ROLE_ARN`).

Reply:
Good catch. I've refactored the code into a separate function so all uses do the same thing.